enterprise/views.py

```diff
@@ -486,6 +489,15 @@ def get_enterprise_course_enrollment_page(self, request, enterprise_uuid, course
             logger.error('Unable to find enterprise customer for UUID: %s', enterprise_uuid)
             raise Http404
 
+        if EnterpriseCourseEnrollment.objects.filter(
+                enterprise_customer_user__enterprise_customer=enterprise_customer,
+                enterprise_customer_user__user_id=request.user.id,
+                course_id=course_id
+        ).exists():
```

Comment:
Then the user is arriving at the Enterprise enrollment page with a specific course in mind (i.e., they're going to it via an employer LMS), but their enrollment hasn't been tied to the Enterprise. By looking specifically for an Enterprise enrollment, we're able to make the enrollment/Enterprise connection to existing enrollments. I think that's the correct behavior in this case.